qcodes/dataset/measurements.py

```diff
@@ -32,8 +32,10 @@ class ParameterTypeError(Exception):
 
 def is_number(thing: Any) -> bool:
     """
-    Test if an object can be converted to a number
+    Test if an object can be converted to a number UNLESS it is a string
     """
+    if isinstance(thing, str):
+        return False
```

Comment:
what about the case of `nan`?

Reply:
still, what about the case of nan?

Reply:
Isn't it okay? Do you have anything particular in mind? 

I think it's good that we *don't* have magical strings that are not allowed amongst other strings. Imagine a measurement of some type casting speed benchmark. I have categories `['int', 'float', 'nan', 'none', 'null', 'np.int32']`. I'd really like that to just plot without getting an SQLite error that my `nan` (now no longer `"nan"`) can not be inserted in a text-valued column.

Reply:
`"nan"` is not a magical string. The thing is that `float('nan')` returns `nan` (`type(float('nan'))` is `float`). Hence when you are doing `isinstance(thing, str)` before `float(thing)` you are neglecting the case where a string`"nan"` means `nan` floating value. Have a look at converter/adapter functions in sqlite_base - that `"nan"` case is covered there as well.

(additional, not really useful comment: exceptions are cheap in python, so it could be the `try: float(thing)` in case `thing` is a string is faster than `isinstance(thing, str)` :) )

Reply:
and why is this change actually needed?